src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -0,0 +1,1054 @@
+}
+
+template <typename R>
+std::shared_ptr<ExpressionResult<R>> ExpressionEvaluator::_evaluate_like_expression(
+    const BinaryPredicateExpression& expression) {
+  Fail("Can only evaluate predicates to int32_t (aka bool)");
+}
+
+template <>
+std::shared_ptr<ExpressionResult<int32_t>> ExpressionEvaluator::_evaluate_is_null_expression<int32_t>(
+    const IsNullExpression& expression) {
+  std::vector<int32_t> result_values;
+
+  _resolve_to_expression_result_view(*expression.operand(), [&](const auto& view) {
+    result_values.resize(view.size());
+
+    if (expression.predicate_condition == PredicateCondition::IsNull) {
+      for (auto chunk_offset = ChunkOffset{0}; chunk_offset < view.size(); ++chunk_offset) {
+        result_values[chunk_offset] = view.null(chunk_offset);
+      }
+    } else {  // PredicateCondition::IsNotNull
+      for (auto chunk_offset = ChunkOffset{0}; chunk_offset < view.size(); ++chunk_offset) {
+        result_values[chunk_offset] = !view.null(chunk_offset);
+      }
+    }
+  });
+
+  return std::make_shared<ExpressionResult<int32_t>>(std::move(result_values));
+}
+
+template <typename T>
+std::shared_ptr<ExpressionResult<T>> ExpressionEvaluator::_evaluate_is_null_expression(
+    const IsNullExpression& expression) {
+  Fail("Can only evaluate predicates to int32_t (aka bool)");
+}
+
+template <>
+std::shared_ptr<ExpressionResult<int32_t>> ExpressionEvaluator::_evaluate_in_expression<int32_t>(
+    const InExpression& in_expression) {
+  const auto& left_expression = *in_expression.value();
+  const auto& right_expression = *in_expression.set();
+
+  std::vector<int32_t> result_values;
+  std::vector<bool> result_nulls;
+
+  if (right_expression.type == ExpressionType::List) {
+    const auto& array_expression = static_cast<const ListExpression&>(right_expression);
+
+    /**
+     * To keep the code simple for now, transform the InExpression like this:
+     * "a IN (x, y, z)"   ---->   "a = x OR a = y OR a = z"
+     *
+     * But first, out of array_expression.elements(), pick those expressions whose type can be compared with
+     * in_expression.value() so we're not getting "Can't compare Int and String" when doing something crazy like
+     * "5 IN (6, 5, "Hello")
+     */
+    const auto left_is_string = left_expression.data_type() == DataType::String;
+    std::vector<std::shared_ptr<AbstractExpression>> type_compatible_elements;
+    for (const auto& element : array_expression.elements()) {
+      if ((element->data_type() == DataType::String) == left_is_string) {
+        type_compatible_elements.emplace_back(element);
+      }
+    }
+
+    if (type_compatible_elements.empty()) {
+      // `5 IN ()` is FALSE as is `NULL IN ()`
+      return std::make_shared<ExpressionResult<int32_t>>(std::vector<int32_t>{0});
+    }
+
+    std::shared_ptr<AbstractExpression> predicate_disjunction =
+        equals(in_expression.value(), type_compatible_elements.front());
```

Comment:
I really don't like this. A method `equals()` should never NOT return a bool. First, brace-init would show that we are creating an object here. Second, let's get rid of `using namespace opossum::expression_factory;` and write `expression_factory::equals` here.

Reply:
suffix `_`